api/src/controllers/login.js

```diff
@@ -182,18 +175,16 @@ const setCookies = (req, res, sessionRes) => {
         .getUserSettings(userCtx)
         .catch(err => {
           if (err.status === 404 && auth.isDbAdmin(userCtx)) {
-            return users.createAdmin(userCtx).then(() => auth.getUserSettings(userCtx));
+            return users.createAdmin(userCtx);
```

Comment:
I is possible that this call to `getUserSettings` on line 175 is unnecessary now (since the actually settings do not get used here).  However, I have left it in since the 404 for that call is what triggers the `createAdmin` logic. If there is a better way to check that a DbAdmin needs to be initialized, we could do that here instead.

Reply:
Either way, it seemed that the second call to getUserSettings (to retrieve the newly created admin's settings) was useless now, so I removed it.

Reply:
I agree. If we don't use `userSettings` anymore, we shouldn't request them. 
We should refactor `createAdmin` to deal with existent accounts and create new ones. We could just keep the one condition here: 
```
if (auth.isDbAdmin(userCtx)) {
  return users.createAdmin(userCtx);
}
```